app/org/sagebionetworks/bridge/validators/SurveySaveValidator.java

```diff
@@ -224,18 +263,30 @@ private void validateOneRuleSet(Errors errors, SurveyRule rule, Set<String> alre
         if (rule.getAssignDataGroup() != null) {
             actionCount++;
         }
-        if (actionCount != 1) {
-            errors.rejectValue(fieldPath, "must have one and only one action: skipTo, endSurvey, or assignDataGroup");
+        if (rule.getDisplayIf() != null) {
+            actionCount++;
         }
-        if (rule.getAssignDataGroup() != null && !dataGroups.contains(rule.getAssignDataGroup())) {
-            errors.rejectValue(fieldPath, "has a data group '" + rule.getAssignDataGroup()
-                    + "' that is not a valid data group: " + BridgeUtils.COMMA_SPACE_JOINER.join(dataGroups));            
+        if (rule.getDisplayUnless() != null) {
+            actionCount++;
         }
```

Comment:
It doesn't appear that we have any tests that verify that you can't have displayIf and displayUnless at the same time. We also want to avoid having 2^5 tests testing every possible combination, as that would not be maintainable.

My recommendation is to have N tests, one for each action, so that we can verify that they're each being counted by hasOneAction(), as well as a test with no actions.